### PR TITLE
RFC-0172 - Technical Editing

### PR DESCRIPTION
**Pull request by anselld**
## Description
Technical editing.

## Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [ ] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Feature refactor (No new feature or functional changes, but performance or technical debt improvements)
* [ ] New Tests
* [x] Documentation

## Checklist:

* [x] I'm merging against the `development` branch
* [ ] I ran `cargo-fmt --all` before pushing
* [ ] My change requires a change to the documentation.
* [ ] I have updated the documentation accordingly.
* [ ] I have added tests to cover my changes.

### CODE REVIEW

**Comment by anselld** [comment]
Hi @hansieodendaal, if you get a chance, please review my changes. I checked all the links and was able to fix most. However, I am having problems with the links for'MessageEnvelope', 'Messageheader' and 'MessageBody'. I would appreciate it if you could please let me know what I'm doing wrong. Thanks!

**Comment by anselld** [comment]
Hi @hansieodendaal, sorry about the accidental closing of this PR. I've reopened it.

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 1]
```diff
@@ -222,41 +224,42 @@ This illustrates the structure of a Tari message.
...
-| body     | 3     | varies          | `Vec<u8>` | Serialized bytes of data containing a unencrypted or encrypted [MessageEnvelopeBody].                                                                      |
+| header   | 2     | Varies          | `Vec<u8>` | Serialized bytes of data containing an unencrypted [MessageEnvelopeHeader].                                                                                      |
+| body     | 3     | Varies          | `Vec<u8>` | Serialized bytes of data containing an unencrypted or encrypted [MessageEnvelopeBody].                                                                      |
 
-The header and decrypted body MUST be deserializable as per [RFC-0171: Message Serialisation](RFC-0171_MessageSerialisation.md)
+The header and decrypted body MUST be deserializable as per [RFC-0171: MessageSerialization](RFC-0171_MessageSerialisation.md).
 
 #### MessageEnvelopeHeader
 
 Every [MessageEnvelope] MUST have an unencrypted header containing the following fields:
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
Every MessageEnvelope MUST have an unencrypted header containing the following fields:
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 2]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
-- MUST connect to a TCP endpoint
-  - For an IP [NetAddress], connect to the given host IP and port
-  - For an Onion [NetAddress], connect to the onion address using the tcp protocol (e.g. `tcp://xyz...123.onion:1234`)
-  - For an I2P [NetAddress], as yet undetermined
+  - Using [ZeroMQ], this means setting the socketopts `ZMQ_CURVE_SERVERKEY`, `ZMQ_CURVE_SECRETKEY` and `ZMQ_CURVE_PUBLICKEY`.
+- MUST connect to a TCP endpoint.
+  - For an IP [NetAddress], connect to the given host IP and port.
+  - For an Onion [NetAddress], connect to the onion address using the TCP, e.g. `tcp://xyz...123.onion:1234`.
+  - For an I2P [NetAddress], as yet undetermined.
 - MUST write the parts of the given [MessageEnvelope] to the socket as a multipart message consisting of, in order:
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST write the parts of the given MessageEnvelope to the socket as a multipart message consisting of, in order:
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 3]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
-- MUST listen for data on the given [NetAddress] using an [InboundConnection]
-- MUST sequentially try to connect to one of the peer's [NetAddress]es until one succeeds or all fail using an [OutboundConnection]
-- MUST immediately reject and dispose of a multipart message not consisting of four parts, as detailed in [MessageEnvelope].
-- MUST construct a [MessageEnvelope] from the multiple parts.
-- MUST pass the constructed [MessageEnvelope] to the message handler.
-- Should a connection drop, the connection state MUST transition to `Connecting` and the connection retried.
-- When a shutdown signal is received, MUST send a `net::Disconnect` message and drop the connection.
+- MUST listen for data on the given [NetAddress] using an [InboundConnection];
+- MUST sequentially try to connect to one of the peer's [NetAddress]es until one succeeds or all fail using an [OutboundConnection];
+- MUST immediately reject and dispose of a multipart message not consisting of four parts, as detailed in [MessageEnvelope];
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST immediately reject and dispose of a multipart message not consisting of four parts, as detailed in MessageEnvelope;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 4]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST construct a [MessageEnvelope] from the multiple parts;
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST construct a MessageEnvelope from the multiple parts;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 5]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST pass the constructed [MessageEnvelope] to the message handler;
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST pass the constructed MessageEnvelope to the message handler;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 6]
```diff
@@ -609,32 +610,32 @@ send messages to the rest of the network.
 
 In particular, it is responsible for:
 
-- serializing the message body
-- constructing the [MessageEnvelope]
-- executing the required BroadcastStrategy
-- sending messages using the [ConnectionManager]
+- serializing the message body;
+- constructing the [MessageEnvelope];
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- constructing the MessageEnvelope;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 7]
```diff
@@ -698,12 +699,13 @@ If the HWM is hit:
 [communication node]: Glossary.md#communication-node
 [connectioncontext]: #connectioncontext
 [controlservice]: #controlservice
-[MessageEnvelope]: #MessageEnvelope
-[MessageEnvelopebody]: #MessageEnvelopebody
-[MessageEnvelopeHeader]: #MessageEnvelopeHeader
 [duplicatemessagewindow]: #duplicatemessagewindow
 [inboundconnection]: #inboundconnection
 [message]: #message
+[MessageEnvelope]: #MessageEnvelope
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 8]
```diff
@@ -266,19 +269,19 @@ It consists of a [MessageHeader] and [Message] of a particular predefined [Messa
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]: Please remove square brackets here:
`A MessageEnvelopeBody is the payload of the [MessageEnvelope]. A [MessageEnvelopeBody] may be encrypted as required.` 
to
`A MessageEnvelopeBody is the payload of the MessageEnvelope. A MessageEnvelopeBody may be encrypted as required.`

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 9]
```diff
@@ -266,19 +269,19 @@ It consists of a [MessageHeader] and [Message] of a particular predefined [Messa
 
 #### MessageType
 
-An enumeration of the messages which are part of the Tari network. [MessageType]s are represented
-as an unsigned 8-bit integer and each value must be mapped to a corresponding [Message] struct.
+An enumeration of the messages that are part of the Tari network. [MessageType]s are represented
+as an unsigned eight-bit integer and each value must be mapped to a corresponding [Message] struct.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
as an unsigned eight-bit integer and each value must be mapped to a corresponding Message struct.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 10]
```diff
@@ -266,19 +269,19 @@ It consists of a [MessageHeader] and [Message] of a particular predefined [Messa
...
 
 All [MessageType]s fall within a particular numerical range according to the message's concern:
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
All MessageTypes fall within a particular numerical range according to the message's concern:
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 11]
```diff
@@ -266,19 +269,19 @@ It consists of a [MessageHeader] and [Message] of a particular predefined [Messa
...
+| Category     | Range   | # Message Types | Description                                                  |
+| ------------ | ------- | --------------- | ------------------------------------------------------------ |
+| `reserved`     | 0       | 1               | Reserved for control messages such as `Ack`.                  |
+| `net`        | 1-32    | 32              | Network-related messages such as `join` and `discover`.       |
+| `peer`       | 33-64   | 32              | Peer connection messages, such as `establish connection`.     |
+| `blockchain` | 65-96   | 32              | Messages related to the blockchain, such as `add block`.      |
+| `vn`         | 97-224  | 128             | Messages related to the validator nodes, such as `execute instruction`. |
+| `extended`   | 225-255 | 30              | Reserved for future use.                                      |
 
 In documentation, [MessageType]s can be referred to by the category and name. For example, `peer::EstablishConnection` and
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
In documentation, MessageTypes can be referred to by the category and name. For example, `peer::EstablishConnection` and
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 12]
```diff
@@ -266,19 +269,19 @@ It consists of a [MessageHeader] and [Message] of a particular predefined [Messa
 
 #### MessageType
 
-An enumeration of the messages which are part of the Tari network. [MessageType]s are represented
-as an unsigned 8-bit integer and each value must be mapped to a corresponding [Message] struct.
+An enumeration of the messages that are part of the Tari network. [MessageType]s are represented
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
An enumeration of the messages that are part of the Tari network. MessageTypes are represented
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 13]
```diff
@@ -329,15 +331,15 @@ and short enough to not be a burden on the node.
...
+A thin wrapper around a `ZMQ_ROUTER` socket, which binds to a [NetAddress] and accepts incoming multipart messages.
+This connection blocks until there is data to read, or a timeout is reached. In both cases, the `receive` method
+can be called again (i.e. in a loop) to continue listening for messages. Client code should run this loop in its own thread.
 `send` is only called (if at all) in response to an incoming message.
 
 Fields may include
 
-- a [NetAddress],
-- a timeout,
+- a [NetAddress];
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- a NetAddress;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 14]
```diff
@@ -351,24 +353,24 @@ Methods may include:
 An InboundConnection:
 
 - MUST perform the "server-side" [CurveZMQ](http://curvezmq.org/page:read-the-docs) encryption protocol if encryption is set.
-  - using [ZeroMQ] this means setting the socketopts `ZMQ_CURVE_SERVER` to 1 and `ZMQ_CURVE_SECRETKEY` to the secret key before binding.
+  - Using [ZeroMQ]. this means setting the socketopts `ZMQ_CURVE_SERVER` to 1 and `ZMQ_CURVE_SECRETKEY` to the secret key before binding.
 - MUST listen for and accept TCP connections.
   - For an IP [NetAddress], bind on the given host IP and port.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - For an IP NetAddress, bind on the given host IP and port.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 15]
```diff
@@ -351,24 +353,24 @@ Methods may include:
...
   - For an Onion [NetAddress], bind on 127.0.0.1 and the given port.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - For an Onion NetAddress, bind on 127.0.0.1 and the given port.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 16]
```diff
@@ -351,24 +353,24 @@ Methods may include:
...
   - For an I2P [NetAddress], as yet undetermined.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - For an I2P NetAddress, as yet undetermined.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 17]
```diff
@@ -351,24 +353,24 @@ Methods may include:
...
+A thin wrapper around a `ZMQ_DEALER` socket, which connects to a [NetAddress] and sends outbound multipart messages.
+This connection blocks until data can be written, or a timeout is reached. The timeout should never be reached, as
 [ZeroMQ] internally queues messages to be sent.
 
-Fields may include
+Fields may include:
 
-- a [NetAddress],
-- underlying [ZeroMQ socket],
+- a [NetAddress];
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- a NetAddress;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 18]
```diff
@@ -382,67 +384,68 @@ Methods may include:
 An [OutboundConnection]:
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
An OutboundConnection:
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 19]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
 
 - MUST perform the "client-side" [CurveZMQ](http://curvezmq.org/page:read-the-docs) encryption protocol if encryption is set.
-  - using [ZeroMQ] this means setting the socketopts `ZMQ_CURVE_SERVERKEY`, `ZMQ_CURVE_SECRETKEY` and `ZMQ_CURVE_PUBLICKEY`
-- MUST connect to a TCP endpoint
-  - For an IP [NetAddress], connect to the given host IP and port
-  - For an Onion [NetAddress], connect to the onion address using the tcp protocol (e.g. `tcp://xyz...123.onion:1234`)
-  - For an I2P [NetAddress], as yet undetermined
+  - Using [ZeroMQ], this means setting the socketopts `ZMQ_CURVE_SERVERKEY`, `ZMQ_CURVE_SECRETKEY` and `ZMQ_CURVE_PUBLICKEY`.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - Using ZeroMQ, this means setting the socketopts `ZMQ_CURVE_SERVERKEY`, `ZMQ_CURVE_SECRETKEY` and `ZMQ_CURVE_PUBLICKEY`.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 20]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
+- MUST connect to a TCP endpoint.
+  - For an IP [NetAddress], connect to the given host IP and port.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - For an IP NetAddress, connect to the given host IP and port.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 21]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
+  - For an Onion [NetAddress], connect to the onion address using the TCP, e.g. `tcp://xyz...123.onion:1234`.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - For an Onion NetAddress, connect to the onion address using the TCP, e.g. `tcp://xyz...123.onion:1234`.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 22]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
+  - For an I2P [NetAddress], as yet undetermined.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
  - For an I2P NetAddress, as yet undetermined.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 23]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
-  - version
-  - header
-  - body
-- if specified, MUST set a high water mark (HWM) on the underlying [ZeroMQ] socket
-- If the HWM is reached, a call to `send` MUST return an error and any messages received SHOULD be discarded
+  - identity;
+  - version;
+  - header;
+  - body.
+- If specified, MUST set a High Water Mark (HWM) on the underlying [ZeroMQ] socket.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- If specified, MUST set a High Water Mark (HWM) on the underlying ZeroMQ socket.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 24]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
 
-A Peer may also contain reputation metrics (e.g. rejected_message_count, avg_latency) to be used to decide
+A peer may also contain reputation metrics (e.g. rejected_message_count, avg_latency) to be used to decide
 if a peer should be banned. This mechanism is yet to be decided.
 
 #### PeerConnection
 
-Represents direct bi-directional connection to another node or client. As connections are bi-directional,
+Represents direct bidirectional connection to another node or client. As connections are bidirectional,
 the [PeerConnection] need only hold a single [InboundConnection] or [OutboundConnection], depending on if the
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
the PeerConnection need only hold a single [InboundConnection] or [OutboundConnection], depending on if the
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 25]
```diff
@@ -382,67 +384,68 @@ Methods may include:
...
 
-- Manage the underlying network connections; with automatic reconnection if necessary.
-- Forward incoming messages onto the given handler socket.
-- Send outgoing messages.
+- managing the underlying network connections, with automatic reconnection if necessary;
+- forwarding incoming messages onto the given handler socket; and
+- sending outgoing messages.
 
-Unlike [InboundConnection] and [OutboundConnection] which are essentially stateless,
+Unlike [InboundConnection] and [OutboundConnection], which are essentially stateless,
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
Unlike InboundConnection and OutboundConnection, which are essentially stateless,
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 26]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
 A `PeerConnection`:
 
-- MUST listen for data on the given [NetAddress] using an [InboundConnection]
-- MUST sequentially try to connect to one of the peer's [NetAddress]es until one succeeds or all fail using an [OutboundConnection]
-- MUST immediately reject and dispose of a multipart message not consisting of four parts, as detailed in [MessageEnvelope].
-- MUST construct a [MessageEnvelope] from the multiple parts.
-- MUST pass the constructed [MessageEnvelope] to the message handler.
-- Should a connection drop, the connection state MUST transition to `Connecting` and the connection retried.
-- When a shutdown signal is received, MUST send a `net::Disconnect` message and drop the connection.
+- MUST listen for data on the given [NetAddress] using an [InboundConnection];
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST listen for data on the given [NetAddress] using an InboundConnection;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 27]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST sequentially try to connect to one of the peer's [NetAddress]es until one succeeds or all fail using an [OutboundConnection];
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST sequentially try to connect to one of the peer's NetAddresses until one succeeds or all fail using an OutboundConnection;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 28]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
 
 The `ConnectionManager`:
 
-- MUST call `suspend` on every [PeerConnection] if it's `suspend` method is called
-- MUST call `resume` on every [PeerConnection] if it's `resume` method is called
-- MUST call `shutdown` on every [PeerConnection] if it's `shutdown` method is called
-- MUST create a new [PeerConnection] with the given Peer and NetAddress, when `establish_connection` is called
-- MUST call `shutdown` on the [PeerConnection] and remove the connection for the given Peer, when `disconnect(peer)` is called
-- MAY disconnect peers if the connection has not been used for an extended period
+- MUST call `suspend` on every [PeerConnection] if its `suspend` method is called;
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST call `suspend` on every PeerConnection if its `suspend` method is called;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 29]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST call `resume` on every [PeerConnection] if its `resume` method is called;
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST call `resume` on every PeerConnection if its `resume` method is called;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 30]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST call `shutdown` on every [PeerConnection] if its `shutdown` method is called
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST call `shutdown` on every PeerConnection if its `shutdown` method is called
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 31]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST create a new [PeerConnection] with the given Peer and NetAddress, when `establish_connection` is called;
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST create a new PeerConnection with the given Peer and NetAddress, when `establish_connection` is called;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 32]
```diff
@@ -455,38 +458,38 @@ Methods may include:
...
+- MUST call `shutdown` on the [PeerConnection] and remove the connection for the given peer, when `disconnect(peer)` is called;
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
- MUST call `shutdown` on the PeerConnection and remove the connection for the given peer, when `disconnect(peer)` is called;
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 33]
```diff
@@ -495,62 +498,60 @@ The purpose of this service is to negotiate a new secure PeerConnection.
...
-3. Alice sends an `peer::establish_connection` message, with
-   - the CURVE public key for the socket connection,
-   - the node's public key corresponding to its [Node ID],
+3. Alice sends a `peer::establish_connection` message, with:
+   - the CURVE public key for the socket connection;
+   - the node's public key corresponding to its [Node ID]; and
    - the [NetAddress] of the new PeerConnection.
-4. Bob accepts this request, and opens a new `PeerConnnection` socket using Alice's CURVE public key.
+4. Bob accepts this request and opens a new `PeerConnnection` socket using Alice's CURVE public key.
 5. Bob connects to the given [NetAddress] and sends a `peer::establish_connection` message.
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
5. Bob connects to the given NetAddress and sends a `peer::establish_connection` message.
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 34]
```diff
@@ -646,49 +647,49 @@ The mechanism for this is proposed as follows:
...
-Eventually, the message will reach nodes which either know the destination or are very close to the destination.
+Eventually, the message will reach nodes that either know the destination or are very close to the destination.
 These nodes MUST store the message in some pending message bucket for the destination. The maximum number of
-buckets and the size of each bucket SHOULD be a sufficiently large as to be unlikely to overflow, but not so
+buckets and the size of each bucket SHOULD be sufficiently large as to be unlikely to overflow, but not so
 large as to approach disk space problems. Individual messages should be small and responsibilities for
-storage are spread over the entire network.
+storage spread over the entire network.
 
 A [communication node]
```
Comment (hansieodendaal) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
A communication node
```

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 35]
```diff
@@ -646,49 +647,49 @@ The mechanism for this is proposed as follows:
...
+- When a destination comes online, it will receive pending messages without having to query them.
 - The "closer within a threshold" metric is simple.
-- Messages are stored on multiple peers which makes it less likely for messages to disappear as nodes come and go
+- Messages are stored on multiple peers, which makes it less likely for messages to disappear as nodes come and go
   (depending on threshold $n$).
 
 ### Queue Overflow Strategy
 
-Inbound/OutboundConnections (and therefore PeerConnection) has a high water mark (HWM) set.
+Inbound/OutboundConnections (and therefore PeerConnection) have a High-water Mark (HWM) set.
```
[the remark this answers is not in the document]
Comment (anselld) [reply RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]: Hi Hansie, it should be '... have an HWM set.'

**Comment by anselld** [comment]
Hi @hansieodendaal , please see my comment about 'an HWM'. It should be 'an' because of the 'ay' sound of 'H'. Otherwise, I've committed all your changes. Thanks!

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md (hunk 35 above)
[the remark this answers is not in the document]
Comment (hansieodendaal) [reply RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]: Updated, thanks

File: RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md [hunk 36]
```diff
@@ -646,49 +647,49 @@ The mechanism for this is proposed as follows:
...
```
Comment (anselld) [inline RFC/src/RFC-0172_PeerToPeerMessagingProtocol.md]:
```suggestion
Inbound/OutboundConnections (and therefore PeerConnection) have an HWM set.
```

**Comment by anselld** [comment]
Hi @hansieodendaal , all changes have been committed and commits combined. Thanks for your help.

**Pull request merged**